### ISSUE TITLE
Locking *.timeshift.json does not work as expected

### ISSUE THREAD

**Issue by masnagam** [mirakc/mirakc#228]
As a result, we can see errors like below:

> ERROR mirakc_timeshift_fs::filesystem: ino#0100000300000000: Failed to read timeshift data: JSON error: EOF while parsing a string at line 1 column 524288

mirakc uses file locks in order to prevent reading and writing each `*.timeshift.json` file at the same time.  We expected that `fs2::FileExt::lock_xxx` works among containers which are shared files on the host filesystem.  However, it seems not to work as expected.

**Comment by masnagam** [comment]
At least, `flock` seems to work properly:

```console
# Containers shares files under /media/timeshift which is mounted onto the containers.
root@mirakc:/# flock /media/timeshift/lock sleep 5

# Blocked until unlocked on the mirakc container
root@mirakc-timeshift-fs:/# flock -s /media/timeshift/lock true
```

**Comment by masnagam** [comment]
There are two patterns of errors.

Pattern 1:

>  ERROR mirakc_timeshift_fs::filesystem: ino#0200000160CA9FBC: Failed to read timeshift data: JSON error: EOF while parsing a value at line 1 column 0

Probably, mirakc-timeshift-fs read after the file was truncated by `std::fs::File::create()`, or before the file was flushed.

Pattern 2:

> ERROR mirakc_timeshift_fs::filesystem: ino#100000300000000: Failed to read timeshift data: JSON error: EOF while parsing a string at line 1 column 524288

???

**Comment by masnagam** [comment]
This issue seems to be solved in 805a5f2.
